src/mail.sh

```diff
@@ -0,0 +1,125 @@
+# This file handles all the interactions of git send-email. It handles its
+# configuration proccess to facilitate the sending of patches via e-mail.
+
+include "$KW_LIB_DIR/kwlib.sh"
+include "$KW_LIB_DIR/kwio.sh"
+include "$KW_LIB_DIR/kw_config_loader.sh"
+
+# Hash containing user options
+declare -gA options_values
+
+function kw_mail()
+{
+  parse_mail_options "$@"
+  if [[ "$?" -gt 0 ]]; then
+    complain "${options_values['ERROR']}"
+    return 22 # EINVAL
+  fi
+
+  if [[ "${options_values['SETUP']}" == 1 ]]; then
+    mail_setup
+    exit
+  fi
+
+  return 0
+}
+
+# This function deals with configuring the necessary options to use
+# the git send-email tool to send patches by email
```

Comment:
Improve this description, explain some of the restrictions around this function and other details developers may not figure out from looking at the code for the first time.